src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,426 @@
+            throw new Error(`No tabs open for server id ${serverId}`);
+        }
+        return firstTab;
+    }
+
+    private persistServers = async (lastActiveTeam?: number) => {
+        this.emit(SERVERS_UPDATE);
+
+        const localServers = [...this.servers.values()].
+            filter((server) => !server.isPredefined).
+            map((server) => this.toConfigTeam(server));
+        await Config.setServers(localServers, lastActiveTeam);
+    }
+
+    private getLastActiveTab = (serverId: string) => {
+        let lastActiveTab: number | undefined;
+        if (this.lastActiveTab.has(serverId)) {
+            const index = this.tabOrder.get(serverId)?.indexOf(this.lastActiveTab.get(serverId)!);
+            if (typeof index !== 'undefined' && index >= 0) {
+                lastActiveTab = index;
+            }
+        }
+        return lastActiveTab;
+    }
+
+    private toConfigTeam = (server: MattermostServer): ConfigTeam => {
+        return {
+            name: server.name,
+            url: `${server.url}`,
+            order: this.serverOrder.indexOf(server.id),
+            lastActiveTab: this.getLastActiveTab(server.id),
+            tabs: this.tabOrder.get(server.id)?.reduce((tabs, tabId, index) => {
+                const tab = this.tabs.get(tabId);
+                if (!tab) {
+                    return tabs;
+                }
+                tabs.push({
+                    name: tab?.type,
+                    order: index,
+                    isOpen: tab.isOpen,
+                });
+                return tabs;
+            }, [] as ConfigTab[]) ?? [],
+        };
+    }
+
+    private getTabView = (srv: MattermostServer, tabName: string, isOpen?: boolean) => {
+        switch (tabName) {
+        case TAB_MESSAGING:
+            return new MessagingTabView(srv, isOpen);
+        case TAB_FOCALBOARD:
+            return new FocalboardTabView(srv, isOpen);
+        case TAB_PLAYBOOKS:
+            return new PlaybooksTabView(srv, isOpen);
+        default:
+            throw new Error('Not implemeneted');
+        }
+    }
+
+    private updateServerURL = (serverId: string) => {
+        const server = this.servers.get(serverId);
+        const remoteInfo = this.remoteInfo.get(serverId);
+
+        if (!(server && remoteInfo)) {
+            return false;
+        }
+
+        if (remoteInfo.siteURL && server.url.toString() !== new URL(remoteInfo.siteURL).toString()) {
+            server.updateURL(remoteInfo.siteURL);
+            this.servers.set(serverId, server);
+            return true;
+        }
+        return false;
+    }
+
+    private openExtraTabs = (serverId: string) => {
+        const server = this.servers.get(serverId);
+        const remoteInfo = this.remoteInfo.get(serverId);
+
+        if (!(server && remoteInfo)) {
+            return false;
+        }
+
+        if (!(remoteInfo.serverVersion && Utils.isVersionGreaterThanOrEqualTo(remoteInfo.serverVersion, '6.0.0'))) {
```

Comment:
any reason why we are checking 6.0.0?

Reply:
Before v6.0, the logic to send React Router pushes through the Desktop App (used to dynamically switch between tabs) didn't exist, so if you're using an older server we don't want to enable these extra tabs in the first place.

This will be going away as well when we move to arbitrary tabs.